### DISCUSSION TITLE
Dot indicators for opened apps

### DISCUSSION THREAD

**Discussion by 5antos** [lawand-dot-io/taskbar#111] [Feature requests]
Great experience with the product so far!

I would like to request having a dot indicator for opened apps (just like macOS does it), instead of automatically moving them to the right in the taskbar. This way, it would avoid having the taskbar app icons always changing positions when opened, specially when they are pinned.

**Comment by lawand-dot-io** [comment]
Hello. I'm glad you enjoy Taskbar. I think what you want to achieve is already possible in case the app you're talking about usually has one window only. You can pin the app (using the "Keep in Taskbar" option) and then add the app to the blacklist (by right clicking it's window, or by adding it in Taskbar's preference). 

In this case, the app no longer takes up space in the Taskbar while it's running.

Is this what you needed? Or is a dot indicator for running apps needed as well? 

Cheers

**Reply by JosTheDude** [reply]
In this case I'm fairly sure the OP is referring to how tasks are closed in Taskbar. When I attempt to close a task such as with Cmd+W -- there's no indicator in the bar that's still "running" per-say and not quit.

I think a good solution would be to either have running tasks move to the right side of the bar with dots that indicate that they're still open but with no windows -- or have dots with the icons like how the MacOS dock does it.

**Reply by 5antos** [reply]
> Hello. I'm glad you enjoy Taskbar. I think what you want to achieve is already possible in case the app you're talking about usually has one window only. You can pin the app (using the "Keep in Taskbar" option) and then add the app to the blacklist (by right clicking it's window, or by adding it in Taskbar's preference).
> 
> In this case, the app no longer takes up space in the Taskbar while it's running.
> 
> Is this what you needed? Or is a dot indicator for running apps needed as well?
> 
> Cheers

Unfortunately, blacklisting the apps so they do not move to the right of the taskbar is not the behavior I was looking for, since you can no longer toggle the open state of them by clicking them in the taskbar.

What I was trying to say is that I would like to see taskbar mimic the macOS dock behavior when opening / hiding (⌘+H/W) apps, i.e.:

- Pinned apps do not shift to the right when opened (multiple windows of the same app could possibly be stacked or something similar, unrelated to this suggestion)
- Non-pinned apps, when opened, are displayed on the right of the taskbar, just like it is currently happening
- Opened apps show a dot indicator below them, like the macOS dock

**Reply by 5antos** [reply]
For reference, the Notes app is not pinned, so it shows on the right in the dock, with the dot indicating it's opened, but the Arc app, since it's pinned, is not shifted to the right of the dock but still shows the dot indicating it's opened:

<img width="447" alt="Screenshot 2024-12-02 at 20 16 33" src="https://github.com/user-attachments/assets/a72a3b6e-2ac7-445d-8ea9-12901365c9e7">

**Reply by lawand-dot-io** [reply]
@5antos I understand what you mean. You are asking for those three things:

- Show icons for running apps, similar to the pin icons that already exist
- Show a running indicator dot under both running app icons and pin icons
- Keep apps at the same spot, whether they are pin icons / running app icons / app windows

Right?

**Reply by lawand-dot-io** [reply]
@JosTheDude Thanks for the explanation 👍

**Reply by 5antos** [reply]
@lawand-dot-io if I'm not mistaken, currently all the running apps appear in the right of the taskbar and with a slightly smaller icon than the pinned ones.

My suggestion is basically just to keep the **pinned** apps in the same spot and a running indicator dot under running apps (instead of making their icons smaller), whether they're pinned or not. Non-pinned running apps can be on the right of the taskbar

**Reply by lawand-dot-io** [reply]
@5antos yes exactly, I got your idea. I will close this request as something similar already exists: https://github.com/lawand-dot-io/taskbar/discussions/76

I added your ideas to the original request. Thanks and enjoy Taskbar
